agent/tcs/client/client.go

```diff
@@ -149,30 +147,85 @@ func (cs *clientServer) publishMetrics() {
 
 // publishMetricsOnce is invoked by the ticker to periodically publish metrics to backend.
 func (cs *clientServer) publishMetricsOnce() {
-	metadata, taskMetrics, err := cs.statsEngine.GetInstanceMetrics()
+	// Get the list of objects to send to backend.
+	requests, err := cs.metricsToPublishMetricRequests()
 	if err != nil {
 		log.Warn("Error getting instance metrics", "err", err)
-		return
 	}
 
-	if *metadata.Idle {
-		// Idle instance, send message and return.
-		cs.MakeRequest(ecstcs.NewPublishMetricsRequest(metadata, taskMetrics))
-		return
+	// Make the publish metrics request to the backend.
+	for _, request := range requests {
+		cs.MakeRequest(request)
 	}
+}
 
+// metricsToPublishMetricRequests gets task metrics and converts them to a list of PublishMetricRequest
+// objects.
+func (cs *clientServer) metricsToPublishMetricRequests() ([]*ecstcs.PublishMetricsRequest, error) {
+	metadata, taskMetrics, err := cs.statsEngine.GetInstanceMetrics()
+	if err != nil {
+		return nil, err
+	}
+
+	var requests []*ecstcs.PublishMetricsRequest
+	if *metadata.Idle {
+		fin := true
+		metadata.Fin = &fin
```

Comment:
You can use the helper `aws.Bool(true)` here; it conveys intent slightly more clearly. There might be other places this can be applied.